js/filter.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+window.filter = (function () {
+  var filterContainer = document.querySelector('.map__filters-container');
+  var filterTypeElement = filterContainer.querySelector('#housing-type');
+  var filterPriceElement = filterContainer.querySelector('#housing-price');
+  var filterRoomsElement = filterContainer.querySelector('#housing-rooms');
+  var filterGuestsElement = filterContainer.querySelector('#housing-guests');
+  var filterFeaturesElement = filterContainer.querySelector('#housing-features');
+  var housingFeaturesCheckboxes = document.querySelector('#housing-features').querySelectorAll('.map__checkbox');
+
+  function filterAds(unfilteredData) {
+    var type = filterTypeElement.value;
+    var price = filterPriceElement.value;
+    var rooms = filterRoomsElement.value;
+    var guests = filterGuestsElement.value;
+    var featuresArray = [];
+
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        featuresArray.push(elem.value);
+      }
+    });
+
+    function isCorrectType(ad) {
+      if (type !== 'any') {
+        return ad.offer.type === type;
+      }
+      return ad;
+    }
+
+    function isCorrectPrice(ad) {
+      if (price !== 'any') {
+        var returnProperty = {
+          'middle': function () {
+            return (ad.offer.price >= 10000 && ad.offer.price <= 50000);
+          },
+          'low': function () {
+            return ad.offer.price < 10000;
+          },
+          'high': function () {
+            return ad.offer.price > 50000;
+          },
+        };
+        return returnProperty[price]();
+      }
+      return ad;
+    }
+
+    function isCorrectRooms(ad) {
+      if (rooms !== 'any') {
+        return ad.offer.rooms === +rooms;
+      }
+      return ad;
+    }
+
+    function isCorrectGuest(ad) {
+      if ((+guests === 0 && ad.offer.guests > 2) || (guests === 'any')) {
+        return ad;
+      }
+      return ad.offer.guests === +guests;
+    }
```

Comment:
should return boolean. 

 it should be either 
(guests === 'any') || (String(ad.offer.guests) === guests) 
or 
(guests === 'any') || (String(ad.offer.guests) === guests) ||  (guests === “0” && ad.offer.guests > 2)

though I am not sure wiether we need (guests === “0” && ad.offer.guests > 2)

Reply:
done